cli/flox/doc/include/package-names.md

```diff
@@ -0,0 +1,24 @@
+## Package names
+Packages are organized in a hierarchical structure such that certain packages
+are found at the top level (e.g. `ripgrep`),
+and other packages are found under package sets (e.g. `python310Packages.pip`).
+
+The full name of a package will be something like
+`legacyPackages.x86_64-linux.python310Packages.pip`
```

Comment:
nonblocking: do we expose `legacyPackages` anywhere? If not do we need to document it?

Reply:
It's in the lockfile

Reply:
Hmm that's a little bit user facing. I don't really think of that as human readable though. We certainly don't need to explain what a `floxpkgs` flake is lol

Reply:
To me it makes sense to very briefly document `legacyPackages` with the following thought process:
- We search a portion of the package's name
- Which portion do we search? -> relative path
- Why "relative"? -> it's a chunk of the full path
- The full path looks like this `...`

Reply:
That is a good point. But that makes me realize we call it `path` in `manifest.toml` so that we don't have to explain that, and maybe our man pages should be consistent

Reply:
How about this:
```
For many use cases, the leading `legacyPackages.<system>` is left off.
The remaining portion of the attribute path (e.g. `python310Packages.pip`) is
referred to as the "relative path" or "path" for short.

This is the portion that is searched when you execute a `flox search` command.
The path is also the portion shown by `flox show`.
Finally, the path appears in your manifest after a `flox install`.
```

Reply:
I like that better 👍